From dd444c92cebfdfa32ec8ec8eaffb432842b55610 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:35 -0400
Subject: [ethernet] be2net: fix log messages in lancer FW download path

Message-id: <1410192881-30702-73-git-send-email-ivecera@redhat.com>
Patchwork-id: 92461
O-Subject: [RHEL7.1 PATCH 72/78] be2net: fix log messages in lancer FW download path
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit bb864e07b2ea9c3cb0206973baa79dc5f27d2ee8
Author: Kalesh AP <kalesh.purayil@emulex.com>
Date:   Tue Sep 2 09:56:51 2014 +0530

    be2net: fix log messages in lancer FW download path

    Log messages in the Lancer FW download path have issues such as:
    - a single message spanning multiple lines
    - the success message is logged even in failure cases
    - status codes are already logged in the FW cmd routines
    This patch fixes these issues.

    Signed-off-by: Kalesh AP <kalesh.purayil@emulex.com>
    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 891ad9e..96db825 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -4072,6 +4072,7 @@ static int lancer_fw_download(struct be_adapter *adapter,
 {
 #define LANCER_FW_DOWNLOAD_CHUNK      (32 * 1024)
 #define LANCER_FW_DOWNLOAD_LOCATION   "/prg"
+ struct device *dev = &adapter->pdev->dev;
  struct be_dma_mem flash_cmd;
  const u8 *data_ptr = NULL;
  u8 *dest_image_ptr = NULL;
@@ -4084,16 +4085,14 @@ static int lancer_fw_download(struct be_adapter *adapter,
  u8 change_status;
 
  if (!IS_ALIGNED(fw->size, sizeof(u32))) {
-  dev_err(&adapter->pdev->dev,
-   "FW Image not properly aligned. "
-   "Length must be 4 byte aligned.\n");
+  dev_err(dev, "FW image size should be multiple of 4\n");
   status = -EINVAL;
   goto lancer_fw_exit;
  }
 
  flash_cmd.size = sizeof(struct lancer_cmd_req_write_object)
     + LANCER_FW_DOWNLOAD_CHUNK;
- flash_cmd.va = dma_alloc_coherent(&adapter->pdev->dev, flash_cmd.size,
+ flash_cmd.va = dma_alloc_coherent(dev, flash_cmd.size,
        &flash_cmd.dma, GFP_KERNEL);
  if (!flash_cmd.va) {
   status = -ENOMEM;
@@ -4133,33 +4132,26 @@ static int lancer_fw_download(struct be_adapter *adapter,
        &add_status);
  }
 
- dma_free_coherent(&adapter->pdev->dev, flash_cmd.size, flash_cmd.va,
-     flash_cmd.dma);
+ dma_free_coherent(dev, flash_cmd.size, flash_cmd.va, flash_cmd.dma);
  if (status) {
-  dev_err(&adapter->pdev->dev,
-   "Firmware load error. "
-   "Status code: 0x%x Additional Status: 0x%x\n",
-   status, add_status);
+  dev_err(dev, "Firmware load error\n");
   goto lancer_fw_exit;
  }
 
+ dev_info(dev, "Firmware flashed successfully\n");
+
  if (change_status == LANCER_FW_RESET_NEEDED) {
-  dev_info(&adapter->pdev->dev,
-    "Resetting adapter to activate new FW\n");
+  dev_info(dev, "Resetting adapter to activate new FW\n");
   status = lancer_physdev_ctrl(adapter,
           PHYSDEV_CONTROL_FW_RESET_MASK);
   if (status) {
-   dev_err(&adapter->pdev->dev,
-    "Adapter busy for FW reset.\n"
-    "New FW will not be active.\n");
+   dev_err(dev, "Adapter busy, could not reset FW\n");
+   dev_err(dev, "Reboot server to activate new FW\n");
    goto lancer_fw_exit;
   }
  } else if (change_status != LANCER_NO_RESET_NEEDED) {
-  dev_err(&adapter->pdev->dev,
-   "System reboot required for new FW to be active\n");
+  dev_info(dev, "Reboot server to activate new FW\n");
  }
-
- dev_info(&adapter->pdev->dev, "Firmware flashed successfully\n");
 lancer_fw_exit:
  return status;
 }
-- 
1.7.1